internal/security/secretstore/init.go

```diff
@@ -59,132 +62,159 @@ func NewBootstrap(insecureSkipVerify bool, vaultInterval int) *Bootstrap {
+		return false
+	}
+
+	lc.Info("SecretStoreClient created")
+
 	pipedHexReader := pipedhexreader.NewPipedHexReader()
-	kdf := kdf.NewKdf(fileOpener, configuration.SecretService.TokenFolderPath, sha256.New)
-	vmkEncryption := NewVMKEncryption(fileOpener, pipedHexReader, kdf)
+	keyDeriver := kdf.NewKdf(fileOpener, secretStoreConfig.TokenFolderPath, sha256.New)
+	vmkEncryption := NewVMKEncryption(fileOpener, pipedHexReader, keyDeriver)
 
 	hook := os.Getenv("IKM_HOOK")
 	if len(hook) > 0 {
 		err := vmkEncryption.LoadIKM(hook)
 		defer vmkEncryption.WipeIKM() // Ensure IKM is wiped from memory
 		if err != nil {
-			lc.Error(fmt.Sprintf("failed to setup vault master key encryption: %s", err.Error()))
+			lc.Errorf("failed to setup vault master key encryption: %s", err.Error())
 			return false
 		}
 		lc.Info("Enabled encryption of Vault master key")
 	} else {
 		lc.Info("vault master key encryption not enabled. IKM_HOOK not set.")
 	}
 
-	var initResponse secretstoreclient.InitResponse // reused many places in below flow
+	var initResponse types.InitResponse // reused many places in below flow
 
 	//step 3: initialize and unseal Vault
 	for shouldContinue := true; shouldContinue; {
 		// Anonymous function used to prevent file handles from accumulating
 		successful := func() bool {
-			sCode, _ := vc.HealthCheck()
+			sCode, _ := client.HealthCheck()
 
 			switch sCode {
 			case http.StatusOK:
 				// Load the init response from disk since we need it to regenerate root token later
-				if err := loadInitResponse(lc, fileOpener, configuration.SecretService, &initResponse); err != nil {
-					lc.Error(fmt.Sprintf("unable to load init response: %s", err.Error()))
+				if err := loadInitResponse(lc, fileOpener, secretStoreConfig, &initResponse); err != nil {
+					lc.Errorf("unable to load init response: %s", err.Error())
 					return false
 				}
-				lc.Info(fmt.Sprintf("vault is initialized and unsealed (status code: %d)", sCode))
+				lc.Infof("vault is initialized and unsealed (status code: %d)", sCode)
 				shouldContinue = false
+
 			case http.StatusTooManyRequests:
-				lc.Error(fmt.Sprintf("vault is unsealed and in standby mode (Status Code: %d)", sCode))
+				// we're done here. Will go into ready mode or reseal
 				shouldContinue = false
+
 			case http.StatusNotImplemented:
-				lc.Info(fmt.Sprintf("vault is not initialized (status code: %d). Starting initialization and unseal phases", sCode))
-				_, err := vc.Init(configuration.SecretService.VaultSecretThreshold,
-					configuration.SecretService.VaultSecretShares, &initResponse)
-				if configuration.SecretService.RevokeRootTokens {
+				lc.Infof("vault is not initialized (status code: %d). Starting initialization and unseal phases", sCode)
+				initResponse, err = client.Init(secretStoreConfig.VaultSecretThreshold, secretStoreConfig.VaultSecretShares)
+				if err != nil {
+					lc.Errorf("Unable to Initialize Vault: %s. Will try again...", err.Error())
+					return true
```

Comment:
error message and return true don't match well; the return value of this anonymous function determines whether vault been init'ed successfully or not but the message seems to indicate to retry it again.  In this case, vault hasn't been successfully init'ed yet.

Reply:
This func is kind of big and important. i suggest refactoring this out into another package and write tests to it later in a separate PR.

Reply:
Agreed, but I feel that is beyond the scope of this refactoring effort, which is focused on the secrets module and use of it. I'll add a TODO so we can address it in a later PR.